netket/optimizer/linear_operator.py

```diff
@@ -67,15 +47,30 @@ class AbstractSMatrix:
 
     """
 
-    sr: SR
-    """Parameters for the solution of the system."""
+    diag_shift: float = 0.00
+    """Diagonal shift added to the S matrix."""
```

Comment:
Do we want to keep this here in the base class?

Reply:
We definitely want `diag_shift` in some class that all the implementations subclass. We can introduce a `QGTBase` class that extends `LinearOperator` with this. Whether it's worth doing this depends on whether we'd use `LinearOperator` in other contexts

Reply:
for now I guess we can keep It here